### PR TITLE
[incubator-kie-drools#5711] Broken OOPath expressions

### PR DESCRIPTION
**Pull request by tkobayas**
**Issue**: 
* https://github.com/apache/incubator-kie-drools/issues/5711

<details>
<summary>
How to replicate CI configuration locally?
</summary>

Build Chain tool does "simple" maven build(s), the builds are just Maven commands, but because the repositories relates and depends on each other and any change in API or class method could affect several of those repositories there is a need to use [build-chain tool](https://github.com/kiegroup/github-action-build-chain) to handle cross repository builds and be sure that we always use latest version of the code for each repository.
 
[build-chain tool](https://github.com/kiegroup/github-action-build-chain) is a build tool which can be used on command line locally or in Github Actions workflow(s), in case you need to change multiple repositories and send multiple dependent pull requests related with a change you can easily reproduce the same build by executing it on Github hosted environment or locally in your development environment. See [local execution](https://github.com/kiegroup/github-action-build-chain#local-execution) details to get more information about it.
</details>

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

- for <b>pull request and downstream checks</b>  
  - Push a new commit to the PR. An empty commit would be enough.

- for a <b>full downstream build</b>
  - for <b>github actions</b> job: add the label `run_fdb`

- for <b>Jenkins PR check only</b>
  - If you are an ASF committer for KIE podling, login to Jenkins (https://ci-builds.apache.org/job/KIE/job/drools/), go to the specific PR job, and click on `Build Now` button.
</details>

### CODE REVIEW

**Comment by tkobayas** [comment]
`org.drools.model.codegen.execmodel.QueryTest#testQueryWithOOPath` and `org.drools.model.codegen.execmodel.OOPathTest` are fixed.

Before PR
`drools-model-codegen`
```
[ERROR] Tests run: 2433, Failures: 275, Errors: 0, Skipped: 9
```

After PR
`drools-model-codegen`
```
[ERROR] Tests run: 2433, Failures: 243, Errors: 0, Skipped: 9

**Comment by tkobayas** [comment]
@yurloc @mariofusco @gitgabrio Please review, thanks!

**Review by mariofusco — APPROVED** [review]

File: drools-drl/drools-drl-parser/src/main/antlr4/org/drools/drl/parser/antlr4/DRLParser.g4 [hunk 1]
```diff
@@ -127,7 +127,7 @@ lhsPattern : xpathPrimary (OVER patternFilter)? |
 lhsPattern : QUESTION? objectType=drlQualifiedName LPAREN positionalConstraints? constraints? RPAREN drlAnnotation* (DRL_OVER patternFilter)? (DRL_FROM patternSource)? ;
 positionalConstraints : constraint (COMMA constraint)* SEMI ;
 constraints : constraint (COMMA constraint)* ;
-constraint : ( nestedConstraint | conditionalOrExpression ) ;
+constraint : ( nestedConstraint | conditionalOrExpression | xpathPrimary) ;
```
Comment (yurloc) [inline drools-drl/drools-drl-parser/src/main/antlr4/org/drools/drl/parser/antlr4/DRLParser.g4]: Not here but on line 253 like:
```
drlExpression
    : drlPrimary
    | xpathPrimary
    ...
```

**Review by yurloc — CHANGES_REQUESTED** [review]
I believe we should 100% mimic `Drl6Expressions.g4` when it comes to constraints to preserve the same behavior as the old parser.

For example, the old parser allows nesting ooPaths and combining them with other expressions using logical operators like this:

```
 Man( /wife[$age : age] && age > $age )
```
This is not possible with the current proposal, which forces the constraint to choose exclusively between either nested constraint, conditional expression, or oopath.

File: drools-drl/drools-drl-parser/src/main/antlr4/org/drools/drl/parser/antlr4/DRLParser.g4 (hunk 1 above)
Comment (tkobayas) [reply drools-drl/drools-drl-parser/src/main/antlr4/org/drools/drl/parser/antlr4/DRLParser.g4]: @yurloc Thank you very much for the point! Yes, I fixed it to accept the mixture of OOPath and standard constraint with a unit test.

**Review by yurloc — APPROVED** [review]
LGTM. Thank you!

**Comment by yurloc** [comment]
Overall fixing progress: There were 290 failed tests.

**Pull request merged**